src/kohlrahbi/__init__.py

```diff
@@ -104,56 +106,93 @@ def load_all_known_pruefis_from_file(
+def create_sheet_name(filename: str):
     """
-    A program to get a machine readable version of the AHBs docx files published by edi@energy.
+    Creates a sheet name from the filename.
+
+    We need to shorten the sheet name because Excel only allows 31 characters for sheet names.
+    This function replaces some words with acronyms and removes some words.
     """
-    check_python_version()
+    sheet_name = filename.split("-informatorischeLesefassung")[0]
+
+    if "Entscheidungsbaum-Diagramm" in sheet_name:
+        sheet_name = sheet_name.replace("Entscheidungsbaum", "EBDs")
+    if "Artikelnummern" in sheet_name:
+        sheet_name = sheet_name.replace("Artikelnummern", "Artikelnr")
+    if "Codeliste" in sheet_name:
+        sheet_name = sheet_name.replace("Codeliste", "CL")
+    if len(sheet_name) > 31:
+        # Excel only allows 31 characters for sheet names
+        # but REQOTEQUOTESORDERSORDRSPORDCHGAHB is 33 characters long
+        sheet_name = sheet_name.replace("HG", "")
+    return sheet_name
+
+
+def scrape_change_histories(input_path: Path, output_path: Path):
+    """
+    starts the scraping process of the change histories
+    """
+    logger.info("start looking for change histories")
+    docx_file_finder = DocxFileFinder.from_input_path(input_path=input_path)
+    ahb_file_paths: list[Path] = docx_file_finder.get_all_docx_files_which_contain_change_histories()
 
-    if not assume_yes:
-        check_output_path(path=output_path)
-    else:
-        if output_path.exists():
-            click.secho(f"The output directory '{output_path}' exists already.", fg="yellow")
-        else:
-            output_path.mkdir(parents=True)
-            click.secho(f"I created a new directory at {output_path}", fg="yellow")
+    change_history_collection: dict[str, pd.DataFrame] = {}
+
+    for ahb_file_path in ahb_file_paths:
+        # extract the sheet name from the file name
+        # COMDISAHB-informatorischeLesefassung1.0e_99991231_20240401.docx -> COMDISAHB
+        sheet_name = create_sheet_name(ahb_file_path.name)
+
+        doc = docx.Document(ahb_file_path)  # Creating word reader object.
+        logger.info("🤓 Start reading docx file '%s'", str(ahb_file_path))
+        change_history_table: ChangeHistoryTable = get_change_history_table(document=doc)
+
+        if change_history_table is None:
+            continue
+        change_history_table.sanitize_table()
+
+        change_history_collection[sheet_name] = change_history_table.table
+
+    # add timestamp to file name
+    # there are two timestamps: one with datetime and another one with just date information.
+    # It is handy during debugging to save different versions of the output files with the datetime information.
+    # But in production we only want to save one file per day.
+    # current_timestamp = datetime.now().strftime("%Y-%m-%dT%H-%M-%S")
+    current_timestamp = datetime.now().strftime("%Y-%m-%d")
```

Comment:
kannst du utcnow verwenden oder die zeitzone klar machen? so würde das ja zum selben zeitpunkt online (auf einem utc-server) ein anderes ergebnis liefern als lokal auf einem "deutschen" rechner

Reply:
stimmt ändere ich.

Reply:
done in d657cef61624db3405bdfd43a6f327cfb475eb32